eolearn/io/raster_io.py

```diff
@@ -42,39 +42,43 @@ class BaseRasterIoTask(IOTask, metaclass=ABCMeta):
     def __init__(
         self,
         feature: Feature,
-        folder: str,
+        path: str = ...,  # type: ignore[assignment]
         *,
         filesystem: FS | None = None,
         image_dtype: np.dtype | type | None = None,
         no_data_value: float | None = None,
         create: bool = False,
         config: SHConfig | None = None,
+        folder: str | None = None,
     ):
         """
         :param feature: Feature which will be exported or imported
-        :param folder: A path to a main folder containing all image, potentially in its subfolders. If `filesystem`
-            parameter is defined, then `folder` should be a path relative to filesystem object. Otherwise, it should be
-            an absolute path.
+        :param path: A path to the image(s). If `filesystem` is defined, then `folder` should be given relative to the
+            filesystem object. Otherwise, it should be an absolute path.
         :param filesystem: A filesystem object. If not given it will be initialized according to `folder` parameter.
         :param image_dtype: A data type of data in exported images or data imported from images.
-        :param no_data_value: When exporting this is the NoData value of pixels in exported images.
-            When importing this value is assigned to the pixels with NoData.
+        :param no_data_value: When exporting this is the `NoData` value of pixels in exported images. When importing
+            this value is assigned to the pixels with `NoData`.
         :param create: If the filesystem path doesn't exist this flag indicates to either create it or raise an error.
         :param config: A configuration object with AWS credentials. By default, is set to None and in this case the
             default configuration will be taken.
         """
-        ftype, fname = self.parse_feature(feature)
-        if fname is None:
-            raise ValueError(f"Feature {feature} is not eligible for this task.")
-        self.feature = ftype, fname
+        if folder is not None:
+            warnings.warn("The parameter `folder` has been renamed to `path`.", EODeprecationWarning, stacklevel=3)
+            path = folder
+
+        if path is ...:  # type: ignore[comparison-overlap]
+            raise TypeError(f"{type(self).__name__} is missing a required positional argument 'path'.")
```

Comment:
If `path` has to be set, why not just leave it uninitialized and let the function complain on its own?

Reply:
because then `TiffTask(feature, folder=my_path)` would fail.

Reply:
What if both are allowed to be `None`, since we allow the user to provide the path also in the `execute` part? This way we can just check if the `folder` parameter is used in order to warn the user, and there is no need for any ellipsis magic

Reply:
i used ellipsis instead of `None` because `path` will be required after the renaming is finalized. No real magic in it, just a different default since `path=None` and `path=...` invoke different things in (my) mind.

We could make path part of the execution arguments, but that means that things fail much later :/